### PR TITLE
Extract kubeflow logic from ComponentSpec to KubeflowComponent

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This PR extracts all the kubeflow logic from the `ComponentSpec` to the `KubeflowComponent` so the `ComponentSpec` is a minimal representation of the fondant component specification. I also brought the classes a bit more in line with the `Manifest` class.

This is preparation for the work to deduct the output manifest from the input manifest and component spec only.

### CODE REVIEW

File: fondant/component_spec.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 
 @dataclass
-class KubeflowInput:
+class Argument:
     """
     Kubeflow component input argument
```
Comment (NielsRogge) [inline fondant/component_spec.py:43]:
```suggestion
    Kubeflow component argument
```
I assume it represents both input and output arguments

File: fondant/component_spec.py [hunk 2]
```diff
@@ -196,101 +103,41 @@ def _validate_spec(self) -> None:
...
-            inputs.append(
-                KubeflowInput(
-                    name=arg_name.strip(),
-                    description=arg_info["description"].strip(),
-                    type=python2kubeflow_type[arg_info["type"].strip()],
-                )
-            )
+    @classmethod
+    def from_file(cls, path: str) -> "ComponentSpec":
+        """Load the manifest from the file specified by the provided path"""
```
Comment (NielsRogge) [inline fondant/component_spec.py:112]: Do you mean loading the component spec instead of the manifest here? As the manifest doesn't contain thiings like `input_subsets` and `output_subsets`

File: fondant/component_spec.py [hunk 3]
```diff
@@ -196,101 +103,41 @@ def _validate_spec(self) -> None:
...
-            name=self.yaml_spec["name"],
-            description=self.yaml_spec["description"],
-            image=self.yaml_spec["image"],
-            inputs=inputs,
-            outputs=outputs,
-        )
-
-        return kubeflow_component.specification
+    def to_file(self, path) -> None:
+        """Dump the manifest to the file specified by the provided path"""
```
Comment (NielsRogge) [inline fondant/component_spec.py:118]: Same comment here

File: fondant/component_spec.py [hunk 4]
```diff
@@ -300,58 +147,146 @@ def output_subsets(self) -> t.Mapping[str, ComponentSubset]:
...
+
+    @property
+    def kubeflow_specification(self) -> "KubeflowComponent":
+        return KubeflowComponent.from_fondant_component(self)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self._specification!r}"
+
+
+class KubeflowComponent:
```
Comment (NielsRogge) [inline fondant/component_spec.py:181]:
```suggestion
class KubeflowSpec:
```
To be consistent with `FondantSpec` ;)

File: fondant/component_spec.py [hunk 5]
```diff
@@ -300,58 +147,146 @@ def output_subsets(self) -> t.Mapping[str, ComponentSubset]:
...
+
+    Args:
+        specification: The component specification as a Python dict
+    """
+
+    def __init__(self, specification: t.Optional[dict]) -> None:
+        self._specification = specification
+
+    @classmethod
+    def from_fondant_component(
```
Comment (NielsRogge) [inline fondant/component_spec.py:193]:
```suggestion
    def from_fondant_spec(
```

File: fondant/component_spec.py [hunk 6]
```diff
@@ -300,58 +147,146 @@ def output_subsets(self) -> t.Mapping[str, ComponentSubset]:
...
+        for arg in args:
+            arg_name = arg.name.replace("-", "_").strip()
+            arg_name_cmd = f'--{arg.name.replace("_", "-")}'.strip()
+
+            dumped_args.append(arg_name_cmd)
+            dumped_args.append({"inputValue": arg_name})
+
+        return dumped_args
+
+    def to_file(self, path: t.Union[str, Path]) -> None:
```
Comment (PhilippeMoussalli) [inline fondant/component_spec.py:252]: I noticed that when compiling the kfp pipeline you can pass in directly a json string instead of the yaml file. So might be worth switching over to that approach. But this should be handled in separate PR related when automating the pipeline definition and compilation

**Review by PhilippeMoussalli — APPROVED** [review]
Really nice refactoring :+1:

Thanks!

**Review by NielsRogge — APPROVED** [review]
Thanks for improving!

File: fondant/component_spec.py (hunk 6 above)
Comment (RobbeSneyders) [reply fondant/component_spec.py:252]: Ok, the json is available under `_spec` already, so should be easy to expose.
